### ISSUE TITLE
android build target does not take into account -D__ANDROID_API__ 

### ISSUE THREAD

**Issue by petrovr** [openssl/openssl#5634]
Build rules for android are changed recently :"
   Configurations/*.conf: overhaul Android targets.
    
    Move Android targets to separate file, automate sysroot setup and
    add support for NDK 16.
    
    Reviewed-by: Rich Salz <rsalz@openssl.org>
    (Merged from https://github.com/openssl/openssl/pull/5589)"

The issue is that updated configuration script is not able to parse properly existence of preprocesor flag -D__ANDROID_API__=...
If CROSS_SYSROOT is defined system default for API version 14 
If is not defined : 
...
Config target attributes:
...
cppflags => "-D__ANDROID_API__=28 -isystem ...
...
 cxxflags => "-std=c++11 -pthread -Wa,--noexecstack -mandroid --sysroot=\$(ANDROID_NDK)/platforms/android-28/arch-arm",
...
Recorded environment:
CPPFLAGS =  -D__ANDROID_API__=15
...
Makevars:
CPPFLAGS        =  -D__ANDROID_API__=15
...

Issue is not about the fact that build system is able to parse valid preprocessor flag like D__ANDROID_API__=__ANDROID_API_J__.
It is about definition with numeric argument.

Tested with NDK r17-beta1.

**Comment by petrovr** [comment]
change title - added missing "not"

**Comment by dot-asm** [comment]
I'm adding a note (in #5613) that setting CROSS_SYSROOT and passing -D__ANDROID_API__ are incompatible and unsupported. Since the API level does appear in CROSS_SYSROOT value, it would be likely inappropriate to try to override it.

**Comment by yan12125** [comment]
I guess the issue here is that passing -D__ANDROID_API__=XX to ./Configure works but not if the macro is specified in ```$CPPFLAGS``` (I didn't test myself). Supporting both usages would be a nice feature.

**Comment by levitte** [comment]
You can pass cpp flags via `$CPPFLAGS`, but only if you don't pass any compiler / linker flags on the configuration command line. The following doesn't work (CPPFLAGS is ignored):

```
CPPFLAGS=-DFOO ./config -lextra
```

This works:

```
CPPFLAGS=-DFOO LDLIBS=-lextra ./config
```

**Comment by dot-asm** [comment]
This is addressed in #5613. It was some misunderstanding, sorry about that. Closing.
